From 5557d47cd7667fcc96757d4e81b809fc84b4a7d9 Mon Sep 17 00:00:00 2001
From: Shuguang Gong <shuguang.gong@intel.com>
Date: Mon, 15 Jun 2015 15:59:21 +0800
Subject: [PATCH 430/462] ov2685: improve mipi clk & vbalnk for 2M resolution

CATS test shows, when connecting touch panel, the VGA
video fps can drop to 22fps. When connecting to HDMI
The fps can reach 30fps. This should be caused by performace
issue on hal or GFX. Improving mipi clk and vblank
can give HAL&GFX more process time to improve performance

Change-Id: I93e20cbeb45d658355bd9179aac17791ed0ec611
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-31194
Signed-off-by: Shuguang Gong <shuguang.gong@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/#/c/60546/
Reviewed-on: https://icggerrit.ir.intel.com/60546
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: IMIN (SC)
Approver: Jing Tao <jing.tao@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov2685.h | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h
index 9da183d..7a253e3 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h
@@ -318,10 +318,10 @@ static struct ov2685_res_struct ov2685_res[] = {
 	.res	= OV2685_RES_2M,
 	.width	= 1600,
 	.height	= 1200,
-	.pixels_per_line = 1700,
-	.lines_per_frame = 1294,
+	.pixels_per_line = 1664,
+	.lines_per_frame = 1682,
 	.fps	= 30,
-	.pix_clk = 66,
+	.pix_clk = 84,
 	.used	= 0,
 	.regs	= NULL,
 	.skip_frames = 0,
@@ -479,7 +479,7 @@ static struct ov2685_reg const ov2685_2M_init[] = {
 	{OV2685_8BIT, 0x3018 , 0x84},
 	{OV2685_8BIT, 0x301d , 0xf0},
 	{OV2685_8BIT, 0x3020 , 0x00},
-	{OV2685_8BIT, 0x3082 , 0x37},/*mclk = 19.2Mhz*/
+	{OV2685_8BIT, 0x3082 , 0x46},/*mclk = 19.2Mhz*/
 	{OV2685_8BIT, 0x3083 , 0x03},
 	{OV2685_8BIT, 0x3084 , 0x07},
 	{OV2685_8BIT, 0x3085 , 0x03},
@@ -524,9 +524,9 @@ static struct ov2685_reg const ov2685_2M_init[] = {
 	{OV2685_8BIT, 0x380a , 0x04},
 	{OV2685_8BIT, 0x380b , 0xb0},
 	{OV2685_8BIT, 0x380c , 0x06},
-	{OV2685_8BIT, 0x380d , 0xa4},
-	{OV2685_8BIT, 0x380e , 0x05},
-	{OV2685_8BIT, 0x380f , 0x0e},
+	{OV2685_8BIT, 0x380d , 0x80},
+	{OV2685_8BIT, 0x380e , 0x06},
+	{OV2685_8BIT, 0x380f , 0x92},
 	{OV2685_8BIT, 0x3810 , 0x00},
 	{OV2685_8BIT, 0x3811 , 0x08},
 	{OV2685_8BIT, 0x3812 , 0x00},
-- 
1.9.1

